CMakeLists.txt

```diff
@@ -15,54 +15,89 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
-find_package(GDAL 3 REQUIRED)
 find_package(ament_cmake REQUIRED)
+find_package(grid_map_msgs REQUIRED)
 find_package(grid_map_ros REQUIRED)
 find_package(grid_map_core REQUIRED)
-find_package(Eigen3 REQUIRED)
+find_package(rclcpp REQUIRED)
 
-add_library(${PROJECT_NAME})
-add_subdirectory(src)
+find_package(Eigen3 REQUIRED)
+find_package(GDAL 3 REQUIRED)
 
-target_include_directories(${PROJECT_NAME}
-    PUBLIC
-    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-    "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
-)
+# macOS
+if(APPLE)
```

Comment:
For cross-compilation purposes, have you considered using `"${CMAK_SYSTEM_NAME} MATCHES Darwin` as a condition?

Reply:
I should use:

```cmake
if(${CMAKE_SYSTEM_NAME} MATCHES Darwin)
  if(${CMAKE_SYSTEM_PROCESSOR} MATCHES arm64)
    link_directories(/opt/homebrew/opt/yaml-cpp/lib)
  endif()
  if(${CMAKE_SYSTEM_PROCESSOR} MATCHES arm64)
    link_directories(/usr/local/opt/yaml-cpp/lib)
  endif()
endif()
```

None of this should be necessarey, but something has gone wrong with the brew `yaml-cpp` or ROS 2 `yaml-cpp-vendored` package so it's dependencies are not propagated.

Reply:
Which version(s) do you have? After this patch I did, the exported namespace targets is available, and worked at least on Ubuntu. The 0.8 release has that included. 

https://github.com/jbeder/yaml-cpp/pull/1196